alluxio/posix/fileimpl.py

```diff
@@ -0,0 +1,219 @@
+        except Exception as e:
+            logging.error(
+                f"Failed to list directory by delegateFileSystem with exception: {e}."
+                f"Used local filesystem instead.")
+            return local_listdir(path, **kw)
+    return local_listdir(path, **kw)
+
+
+def mkdir(path: str, mode=0o777, **kw):
+    instance = DelegateFileSystem.instance
+    fs = instance.get_file_system(path)
+    if fs:
+        try:
+            return fs.mkdir(path, mode, **kw)
+        except Exception as e:
+            logging.error(
+                f"Failed to make directory by delegateFileSystem with exception: {e}."
+                f"Used local filesystem instead.")
+            return local_mkdir(path, mode, **kw)
+    return local_mkdir(path, mode, **kw)
+
+
+def rmdir(path: str, **kw):
+    instance = DelegateFileSystem.instance
+    fs = instance.get_file_system(path)
+    if fs:
+        try:
+            return fs.rmdir(path, **kw)
+        except Exception as e:
+            logging.error(
+                f"Failed to remove directory by delegateFileSystem with exception: {e}."
+                f"Used local filesystem instead."
+            )
+            return local_rmdir(path, **kw)
+    return local_rmdir(path, **kw)
+
+
+def remove(path: str, **kw):
+    instance = DelegateFileSystem.instance
+    fs = instance.get_file_system(path)
+    if fs:
+        try:
+            return fs.rm(path, **kw)
+        except Exception as e:
+            logging.error(
+                f"Failed to remove file by delegateFileSystem with exception: {e}."
+                f"Used local filesystem instead.")
+            return local_remove(path, **kw)
+    return local_remove(path, **kw)
+
+
+def rename(src: str, dest: str, **kw):
+    instance = DelegateFileSystem.instance
+    fs_src = instance.get_file_system(src)
+    fs_dest = instance.get_file_system(dest)
+    if fs_src and fs_dest and fs_src == fs_dest:
+        try:
+            return fs_src.rename(src, dest, **kw)
+        except Exception as e:
+            logging.error(
+                f"Failed to rename file by delegateFileSystem with exception: {e}."
+                f"Used local filesystem instead.")
+            return local_rename(src, dest, **kw)
+    logging.error("Source and destination are on different file systems or not supported.")
+    return local_rename(src, dest, **kw)
+
+
+class DelegateFileSystem:
+    instance = None
+
+    def __init__(self, config_manager: ConfigManager):
+        self.config_manager = config_manager
+        self.filesystem_storage = FSStorage()
+        self.filesystem_storage.data = {}
+        self.enableFileSystems = [Constants.OSS_FILESYSTEM_TYPE,
+                                  Constants.ALLUXIO_FILESYSTEM_TYPE,
+                                  Constants.S3_FILESYSTEM_TYPE]
+        self.__init__file__system()
+        DelegateFileSystem.instance = self
+
+    def __create__file__system(self, fs_name: str):
```

Comment:
Is it impossible to distinguish uniqueness only through fs name? For example, there are multiple buckets under the same ufs.